library/core/src/num/uint_macros.rs

```diff
@@ -1501,6 +1501,33 @@ macro_rules! uint_impl {
             }
         }
 
+                /// Unbounded shift left. Computes `self << rhs`, without bounding the value of `rhs`
+        /// 
+        /// If `rhs` is larger or equal to the number of bits in `self`, 
+        /// the entire value is shifted out, and `0` is returned.
+        ///
+        /// # Examples
+        /// 
+        /// Basic usage:
+        /// ```
+        #[doc = concat!("assert_eq!(0x1", stringify!($SelfT), ".unbounded_shl(4), 0x10);")]
+        #[doc = concat!("assert_eq!(0x1", stringify!($SelfT), ".unbounded_shl(129), 0);")]
+        /// ```
+        #[unstable(feature = "unbounded_shifts", issue = "129375")]
+        #[rustc_allow_const_fn_unstable(unchecked_shifts)]
+        #[must_use = "this returns the result of the operation, \
+                      without modifying the original"]
+        #[inline]
+        pub const fn unbounded_shl(self, rhs: u32) -> $SelfT{
+            if rhs < Self::BITS{
+                // SAFETY:
+                // rhs is just checked to be in-range above
+                unsafe { self.unchecked_shl(rhs) }
+            }else{
+                0
+            }
```

Comment:
`checked_shl` seems to be using `self.unchecked_shl` as well, as does `wrapping_shl`.

It would be nice to get rid of the unneeded `UbCheck`, but I think it would be better as a separate PR with the rest of them (maybe to call the intrinsic directly). I'd rather keep consistency with the rest of the macro for now.

Reply:
Checking on CE, `unwrap_or` isn't `const fn` compatible. I could do a manual match, I suppose.

Reply:
why is `unwrap_or` not a `const fn`...

Reply:
The body isn't valid for the generic `T`, since it requires `Drop` glue for `T`.

Reply:
Yeah a bunch of useful `Option` methods are not `const fn` and blocked on https://github.com/rust-lang/rust/issues/73255, which is blocked on finding someone who both cares and understands our drop elaboration story and how borrowck treats drop...

Reply:
Oh never mind, this one actually *does* drop something. So that's not blocked on `const_precise_live_drops`, it's blocked on way more complicated things where we can declare this function `const fn` only if `T` is `const Drop`...

Reply:
Yep - and `unwrap_or_else` also deals with FnOnce()-in-const